example/src/screens/Tests/tests/urlTests.ts

```diff
@@ -0,0 +1,205 @@
+        url.search = '?test=newValue';
+        expect(url.searchParams.size).to.equal(1);
+        expect(url.searchParams.get('test')).to.equal('newValue');
+      });
+
+      it('should update searchParams if href is set', () => {
+        const url = new URL('http://www.example.com?existingKey=existingValue');
+        url.href = 'http://www.example.com';
+        expect(url.searchParams.size).to.equal(0);
+      });
+
+      it('should reflect changes in searchParams in the relevant properties of the URL', () => {
+        const url = new URL('http://www.example.com');
+        url.searchParams.append('key', 'value');
+
+        // Ensure that each method is updated properky
+        expect(url.search).to.equal('?key=value');
+        expect(url.href).to.equal('http://www.example.com/?key=value');
+        expect(url.toString()).to.equal('http://www.example.com/?key=value');
+        expect(url.toJSON()).to.equal('http://www.example.com/?key=value');
+      });
+
+      it('should update the URL when searchParams are modified', () => {
+        const url = new URL('http://www.example.com?initial=param');
+        url.searchParams.set('initial', 'newvalue');
+        expect(url.search).to.equal('?initial=newvalue');
+      });
+
+      it('should allow modification of existing search parameters through searchParams', () => {
+        const url = new URL('http://www.example.com?existingKey=existingValue');
+        url.searchParams.set('existingKey', 'newValue');
+        expect(url.search).to.equal('?existingKey=newValue');
+      });
+
+      it('should reflect deletion of search parameters in the URL search property', () => {
+        const url = new URL('http://www.example.com?key1=value1&key2=value2');
+        url.searchParams.delete('key1');
+        expect(url.search).to.equal('?key2=value2');
+      });
+
+      it('should handle multiple search parameters correctly', () => {
+        const url = new URL(
+          'http://www.example.com?initialKey=initialValue&anotherKey=anotherValue'
+        );
+        url.searchParams.append('newKey', 'newValue');
+        expect(url.search).to.include('initialKey=initialValue');
+        expect(url.search).to.include('anotherKey=anotherValue');
+        expect(url.search).to.include('newKey=newValue');
+      });
+    });
+
+    describe('ToString and ToJSON Tests', () => {
+      const exampleUrl = 'http://www.example.com/path?query=string#hash';
+
+      it('should return the full URL string using toString', () => {
+        const url = new URL(exampleUrl);
+        expect(url.toString()).to.equal(exampleUrl);
+      });
+
+      it('should return the full URL string using toJSON', () => {
+        const url = new URL(exampleUrl);
+        expect(url.toJSON()).to.equal(exampleUrl);
+      });
+    });
+
+    describe('Static Methods Tests', () => {
+      describe('createObjectURL', () => {
+        // TODO: Currently this test fails with "Cannot create URL for a blob!"
```

Comment:
@KusStar I'm not really sure what is going on here - maybe you can help with explaining how this should work?

Reply:
Oh, it was copied from react-native's URL implementation, but I forgot the comment. I will add it later.

```
 * To allow Blobs be accessed via `content://` URIs,
 * you need to register `BlobProvider` as a ContentProvider in your app's `AndroidManifest.xml`:
```
See https://github.com/facebook/react-native/blob/cfc0ba04a1fa147517e94d1f952e2a8fe62e6e9c/packages/react-native/Libraries/Blob/URL.js#L30-L53